### PR TITLE
BaSP-MR-146: Admins react hook form implementation

### PR DESCRIPTION
**Pull request by ale-schmidt**
Update all current forms, along with the shared components (different types of inputs) used within them, following the documentation provided by React Hook Form. Additionally, the corresponding validations should be added to all forms.

As a final result, all forms in the frontend application should work with the React Hook Form library, maintaining a pleasant and consistent user experience, with all the respective validations for each input type and requirement, reusing components and functionalities.

- **Implement shared input ( HIGH PRIORITY )
- Update all current forms and shared components used in the forms (inputs). Take into account styles for errors
- Adding validations to forms (required fields, dates, names, number of characters). Using the JOI library)
- Add reset function button to clear fields**

### CODE REVIEW

**Review by EmaBaldor — APPROVED** [review]
God job!.
Note the styles so that it shows two columns of inputs.

File: src/validations/admins.jsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    .max(25)
+    .messages({
+      'string.min': 'City must have between 5 and 25 characters',
+      'string.max': 'City must have between 5 and 25 characters'
+    }),
+  password: Joi.string()
+    .regex(/^(?=\w*\d)(?=\w*[A-Z])(?=\w*[a-z])/)
+    .trim()
+    .min(8)
+    .max(20)
```
Comment (marcohzo) [inline src/validations/admins.jsx]: The maximum number of characters must match the server validations (max 16)

**Review by marcohzo — CHANGES_REQUESTED** [review]

**Review by marcohzo — APPROVED** [review]
Good job!

**Comment by ale-schmidt** [comment]
[[Admins] - React-hook-form implementation](https://trello.com/c/kHQ5BD69/146-admins-react-hook-form-implementation)

File: src/Components/Admins/Form/index.jsx [hunk 2]
```diff
@@ -87,118 +91,106 @@ const Form = () => {
...
-              name="password"
-              type="text"
-              value={admin.password}
-              placeholder="Password"
-              onChange={handleOnChange}
-              required
+            <Input
+              register={register}
+              labelName={'Password'}
+              inputType={'text'}
```
Comment (lorenzomauro59) [inline src/Components/Admins/Form/index.jsx:171]:
```suggestion
              inputType={'password'}
```

**Review by lorenzomauro59 — APPROVED** [review]
Only the detail that I commented, rest is working. Good job!!

**Comment by araefedg** [comment]
I found a couple of validations in your bugs, I left you a ticket in Trello: https://trello.com/c/u4YWo8ft

**Pull request merged**